pyccel/codegen/printing/cwrappercode.py

```diff
@@ -273,88 +268,187 @@ def get_cast_function_call(self, cast_type, arg):
+        error_check  : Bool
+            True if checking the data type and raising error is needed
+
+        Returns:
+        -------
+        python_type_check : FunctionCall
+            functionCall responsible of python data type check
+            None when error_check is False
+
+        numpy_type_check : FunctionCall or None
+            functionCall responsible of numpy data type check
+
+        error : FunctionCall or None
+            function call that raise TypeError
+            None when error_check is False
+        """
+
+        if not error_check :
+            scalar_check =  FunctionCall(PyArray_CheckScalar, [collect_var])
+            return scalar_check, LiteralTrue(), None
+
+        numpy_type_check  = NumpyType_Check(variable, collect_var)
+        python_type_check = None
+
+        #When the variable precision is equal to default system precision a python type check is needed
+        if variable.precision == default_precision[str_dtype(variable.dtype)] :
+            python_type_check = PythonType_Check(variable, collect_var)
+
+        error = PyErr_SetString('PyExc_TypeError', self._generate_TypeError_message(variable))
+
+        return numpy_type_check, python_type_check, error
+
     # -------------------------------------------------------------------
     # Functions managing  the creation of wrapper body
     # -------------------------------------------------------------------
-    def _body_optional_variable(self, tmp_variable, variable, collect_var, check_type = False):
+
+    def _valued_variable_management(self, variable, collect_var, tmp_variable):
         """
-        Responsible for collecting value and managing error and create the body
-        of optional arguments in format
-                if (pyobject == Py_None){
-                    collect Null
-                }else if(Type Check == False){
-                    Print TypeError Wrong Type
-                    return Null
-                }else{
-                    assign pyobject value to tmp variable
-                    collect the adress of the tmp variable
-                }
+        Responsible for creating the body collecting the default value of an valuedVariable
+        and the check needed.
+        if the valuedVariable is optional create body to collect the new value
+
         Parameters:
         ----------
         tmp_variable : Variable
             The temporary variable  to hold result
-        Variable : Variable
+        variable     : Variable
             The optional variable
-        collect_var : variable
+        collect_var  : Variable
             the pyobject type variable  holder of value
-        check_type : Boolean
-            True if the type is needed
 
         Returns
         -------
-        body : list
-            A list of statements
+        section      :
+            IfSection
+        collect_body :
+            body that collect new optional variable value
```

Comment:
Grammar : `body that collect` -> `body which collects`

But maybe better:
`list containing the lines necessary to collect the new optional variable value`

Reply:
done 0da2e5b